### ISSUE TITLE
Upgrade from 2.x to 3.x - operation_Name missing from traces, dependencies, and exceptions

### ISSUE THREAD

**Issue by mmcdermottMS** [microsoft/ApplicationInsights-Java#1941]
### Expected behavior
I'm working with a customer to upgrade them from Java Agent v2.x to v3.x in order to improve distributed tracing across a microservices architecture.  They upgraded in a lower environment but are reporting that operation_Name is no longer being populated for **trace**, **dependency**, and **exception** item types.  They also indicated that the operation_Name format has changed, but I indicated that is [by design](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-upgrade-from-2x#operation-names)

I am able to reproduce the behavior in a sandbox environment with a mock up of their architecture.

Is this to be expected with the upgrade from v2.x to v3.x or is this a potential bug?   If the former, is there a way to inject the operation_Name into those data models via code or config?

### Actual behavior
The documentation isn't terribly clear, but I expect to see operation_Name populated for **trace**, **dependency**, and **exception** item types but in the aggregation friendly format new to the v3 agent

### To Reproduce

1. Create at least two basic spring-boot based Java web apps, each with at least 1 controller method (POST or GET, doesn't matter)
2. Configure the applications so that one of the applications makes an HTTP request to the second to simulate API calls between apps
3. Deploy both apps to their own App Service running on a common Linux based App Service Plan
4. Create an Log Analytics Workspace
5. Create an Application Insights instance using the LAW from step 4
6. Enable application insights within the Azure Portal for both deployed Java app services (from step 3) to use the AppInsights instance from step 5
7. Confirm the **ApplicationInsightsAgent_EXTENSION_VERSION** is set to ~3 in the configuration settings for both app services
8. Run some HTTP calls through the 1st app service
9. Check the AI logs and observe that distributed tracing is working and dependent requests captured, but observe that operation_Name is empty for most of the **trace**, **dependency**, and **exception** records associated with any given operation_Id and that it's only consistently populated for **request* item types

### System information
Please provide the following information:
 - SDK Version: **Java v11, spring-boot v2.4.11, Application Insights Agent ~3**
 - OS type and version: **Linux App Service Plan (B1: 1)**
 - Using spring-boot? - **Yes**
 - Additional relevant libraries (with version, if applicable): **N/A**

### Logs
I enabled self-diagnostic logs via [this documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#self-diagnostics) but unsure how to retrieve the log file via Kudu

**Comment by trask** [comment]
> operation_Name is no longer being populated for **trace**, **dependency**, and **exception** item types.

This was just resolved in 3.2.0 👍

> the operation_Name format has changed, but I indicated that is by design

What does operation_Name look like that you are seeing? I want to confirm that it looks like what we are expecting in 3.x, and not an edge case that we missed.

**Comment by trask** [comment]
> Is there a way to force the upgrade to 3.2.0? I have **ApplicationInsightsAgent_EXTENSION_VERSION** set to ~3, can I be more specific? Or is this version something that is rolling out to various tenants over time?

3.2.0 won't make it out to App Services for a while. You can manually wire it up in the meantime if needed (upload the jar file, set the `-javaagent` JVM argument).

> As for operation_Name, here is the format for ~2:
> And the format for ~3:

great, that looks like what we'd expect for 3.x (except for operationName missing on other telemetry which is what's fixed in 3.2.0)

**Comment by mmcdermottMS** [comment]
Ok, thank you for confirming.  I did go down the path of manually uploading the 3.2.1 agent jar file and setting the -javaagent value in the JAVA_OPTS config setting, and then removed app service specific config values and lost all telemetry.  

If I go with that approach, should I keep the OOTB Application Insights config settings (ApplicationInsightsAgent_EXTENSION_VERSION, etc...) and if so, will the runtime use the uploaded agent jar file instead of the backend agent?

**Comment by mmcdermottMS** [comment]
So, I did some additional experimentation and I was able to get the agent jar file deployed and working as expected:

1. I added applicationinsights-agent-3.2.1.jar as well as applicationinsights.json to my deployment config
2. Verified that both were deployed to /home/site/wwwroot alongside my app.jar file
3. Added the _-javaagent:/home/site/wwwroot/applicationinsights-agent-3.2.1.jar_ value to the **JAVA_OPTS** app config setting within Configuration blade in the Azure Portal
4. Removed all other Application Insights specific settings from the Configuration blade
5. Verified that **operation_Name** is now being populated in traces, dependencies, and exceptions
6. Verified that applying any one off Application Insights settings (such as logging level) overrode anything I had set in the applicationinsights.json file

At this point I'm going to close this out but hopefully my documented steps can help out anyone else who may run into the same issue.
